### PR TITLE
Ilb 23 UI api integration for vendor listing

### PR DESCRIPTION
**Pull request by NevinJoseph22**
Unit testing of vendor listing with 100% coverage 
image

### CODE REVIEW

File: src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts [hunk 1]
```diff
@@ -1,21 +1,118 @@
...
 import { VendorListingComponent } from './vendor-listing.component';
+import { IVendor } from '../models/vendor.model';
+import { VendorService } from '../services/vendor.service';
+import { RouterTestingModule } from '@angular/router/testing'; // Import RouterTestingModule
+import { of } from 'rxjs';
+import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
 
-describe('ListingComponent', () => {
+describe('VendorListingComponent', () => {
   let component: VendorListingComponent;
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts]: Remove this. Follow reference shared.

File: src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts [hunk 2]
```diff
@@ -1,21 +1,118 @@
...
+  });
 
-    fixture = TestBed.createComponent(VendorListingComponent);
-    component = fixture.componentInstance;
+  beforeEach(() => {
+    ({ fixture, component } = setup());
     fixture.detectChanges();
   });
 
   it('should create', () => {
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts:83]: use setup()

File: src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts [hunk 3]
```diff
@@ -1,21 +1,118 @@
...
+  });
+
+  it('should fetch and sort vendors by name', () => {
+    component.fetchVendors();
+    expect(component.vendors).toEqual(mockVendors);
+    expect(component.vendors[0].name).toBe('Vendor A');
+    expect(component.vendors[1].name).toBe('Vendor B');
+  });
+
+  it('should display "Pending" tag for unapproved vendors', () => {
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts]: This case is not required

File: src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts (hunk 1 above)
Comment (NevinJoseph22) [reply src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts]: done

File: src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts (hunk 2 above)
Comment (NevinJoseph22) [reply src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts:83]: used setup and modified accordingly

File: src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts (hunk 3 above)
Comment (NevinJoseph22) [reply src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts]: it is removed

**Review by athul-sanjeevan — APPROVED** [review]

File: src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts [hunk 4]
```diff
@@ -1,21 +1,95 @@
...
+    return { fixture, component, spyObj };
+  };
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
-      imports: [VendorListingComponent],
+      declarations: [VendorListingComponent],
+      imports: [HttpClientTestingModule, TableModule, RouterTestingModule],
+      schemas: [CUSTOM_ELEMENTS_SCHEMA],
+      providers: [VendorService],
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts]: is this required?

File: src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts [hunk 5]
```diff
@@ -1,21 +1,95 @@
...
-    component = fixture.componentInstance;
-    fixture.detectChanges();
   });
 
   it('should create', () => {
+    const { component } = setup();
     expect(component).toBeTruthy();
   });
+
+  it('should list all the vendors when the user navige to the page', () => {
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts]: spelling mistake in navigate.
user navigate to the vendor listing screen

Comment (NevinJoseph22) [reply src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts]: corrected

File: src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts (hunk 4 above)
Comment (NevinJoseph22) [reply src/app/features/vendor/vendor-listing/vendor-listing.component.spec.ts]: changes made

**Review by kavyamolr — APPROVED** [review]

**Pull request merged**